### PR TITLE
Fix TCP keepalive defaults

### LINKED ISSUE #3061: Socket keepalive settings disabled by default due to typo?
I noticed that there is code in afsocket which looks like it should be setting a 60 second keepalive interval on the socket:

https://github.com/syslog-ng/syslog-ng/blob/master/modules/afsocket/socket-options-inet.c#L234
```
#if defined(TCP_KEEPTIME) && defined(TCP_KEEPIDLE) && defined(TCP_KEEPCNT)
  self->tcp_keepalive_time = 60;
  self->tcp_keepalive_intvl = 10;
  self->tcp_keepalive_probes = 6;
#endif
```
In practice these lines of code have no effect because the Linux header files do not define `TCP_KEEPTIME`. Looking earlier in the file, it looks like the intention was to test for `TCP_KEEPINTVL` instead, i.e.
```
diff --git a/modules/afsocket/socket-options-inet.c b/modules/afsocket/socket-options-inet.c
index f610a89cc..cc777bc94 100644
--- a/modules/afsocket/socket-options-inet.c
+++ b/modules/afsocket/socket-options-inet.c
@@ -231,7 +231,7 @@ socket_options_inet_new_instance(void)
   socket_options_init_instance(&self->super);
   self->super.setup_socket = socket_options_inet_setup_socket;
   self->super.so_keepalive = TRUE;
-#if defined(TCP_KEEPTIME) && defined(TCP_KEEPIDLE) && defined(TCP_KEEPCNT)
+#if defined(TCP_KEEPINTVL) && defined(TCP_KEEPIDLE) && defined(TCP_KEEPCNT)
   self->tcp_keepalive_time = 60;
   self->tcp_keepalive_intvl = 10;
   self->tcp_keepalive_probes = 6;
```
With the patch above applied the socket sends keepalive messages by default:
```
15:36:56.442303 IP 192.0.2.2.43210 > 192.0.2.1.6514: Flags [S], seq 341681766, win 29200, options [mss 1460,sackOK,TS val 82132 ecr 0,nop,wscale 7], length 0
15:36:56.442334 IP 192.0.2.1.6514 > 192.0.2.2.43210: Flags [S.], seq 219356288, ack 341681767, win 28960, options [mss 1460,sackOK,TS val 82132 ecr 82132,nop,wscale 7], length 0
15:36:56.442347 IP 192.0.2.2.43210 > 192.0.2.1.6514: Flags [.], ack 1, win 229, options [nop,nop,TS val 82132 ecr 82132], length 0
15:37:56.518541 IP 192.0.2.2.43210 > 192.0.2.1.6514: Flags [.], ack 1, win 229, options [nop,nop,TS val 97152 ecr 82132], length 0
15:37:56.518572 IP 192.0.2.1.6514 > 192.0.2.2.43210: Flags [.], ack 1, win 227, options [nop,nop,TS val 97152 ecr 82132], length 0
15:38:56.678541 IP 192.0.2.2.43210 > 192.0.2.1.6514: Flags [.], ack 1, win 229, options [nop,nop,TS val 112192 ecr 97152], length 0
15:38:56.678572 IP 192.0.2.1.6514 > 192.0.2.2.43210: Flags [.], ack 1, win 227, options [nop,nop,TS val 112192 ecr 82132], length 0
15:39:56.838544 IP 192.0.2.2.43210 > 192.0.2.1.6514: Flags [.], ack 1, win 229, options [nop,nop,TS val 127232 ecr 112192], length 0
15:39:56.838576 IP 192.0.2.1.6514 > 192.0.2.2.43210: Flags [.], ack 1, win 227, options [nop,nop,TS val 127232 ecr 82132], length 0
```
While this looks good, I see that the documentation states that keepalive is expected to be disabled by default and the parameters set to 0:
https://www.syslog-ng.com/technical-documents/doc/syslog-ng-open-source-edition/3.25/administration-guide/so-keepalive
https://www.syslog-ng.com/technical-documents/doc/syslog-ng-open-source-edition/3.25/administration-guide/tcp-keepalive-time
https://www.syslog-ng.com/technical-documents/doc/syslog-ng-open-source-edition/3.25/administration-guide/tcp-keepalive-intvl
https://www.syslog-ng.com/technical-documents/doc/syslog-ng-open-source-edition/3.25/administration-guide/tcp-keepalive-probes

It is not clear to me whether this means that the code is working as intended and perhaps these lines of code should instead be deleted?

### PR DESCRIPTION
**Pull request by MrAnno**
Due to a typo, our TCP keepalive defaults were never set:
- `tcp-keepalive-time()`: 60
- `tcp-keepalive-intvl()`: 10
- `tcp-keepalive-probes()`: 6

Prior to this fix, we used the kernel defaults (7200 seconds (2 hours) for tcp_keepalive_time on Linux).
From now on, syslog-ng uses the defaults above.

Fixes #3061

Questions:
- Shouldn't we keep the old behavior? `so-keepalive()` was and still is enabled by default, but with kernel defaults.
Those defaults can be changed system-wide with `sysctl`, so I think it's not worth hard-coding our own defaults.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by MrAnno** [comment]
@kira-syslogng build packages

**Comment by MrAnno** [comment]
@kira-syslogng build packages please

**Comment by kira-syslogng** [comment]
Package Build FAILURE

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Package Build FAILURE

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: configure.ac [hunk 1]
```diff
@@ -1590,6 +1590,14 @@ if test "x$enable_ipv6" = "xauto"; then
         fi
 fi
 
+AC_CHECK_DECLS([TCP_KEEPIDLE, TCP_KEEPINTVL, TCP_KEEPCNT], [], [],
+                [[#include <sys/socket.h>
+                  #include <netinet/in.h>
+                  #include <netinet/tcp.h>]])
+
+if test "x$ac_cv_have_decl_TCP_KEEPIDLE" = "xyes" -a "x$ac_cv_have_decl_TCP_KEEPINTVL" = "xyes" -a "x$ac_cv_have_decl_TCP_KEEPCNT" = "xyes"; then
```
Comment (furiel) [inline configure.ac:1598]: You could use the `action-if-found` for this case.

Comment (MrAnno) [reply configure.ac:1598]: `action-if-found` does not work as documented (autoconf v2.69).
I need all 3 symbols to present, my autoconf version runs `action-if-found` when any of the 3 are available.

**Comment by MrAnno** [comment]
I forgot about CMake, I'm going to add that to the PR. Thanks.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**